esp-hal-common/Cargo.toml

```diff
@@ -132,3 +133,7 @@ debug = [
     "esp32s2?/impl-register-debug",
     "esp32s3?/impl-register-debug",
 ]
+
+log = ["dep:log"]
+defmt = ["dep:defmt"]
+log-defmt = ["defmt"]
```

Comment:
What's the purpose of this feature? Couldn't `cfg(feature = "defmt")` be used everywhere instead?

Reply:
I wanted to separate the format macro and the log output itself. I admit it might not make much sense to do so.

Re. macros, I know what embassy does. I don't exactly understand how or why the first module thing works, so I may end up with that. However, I believe sticking to the noop macros makes sense to avoid unused variable warnings in certain cases.

This is very much work in progress and should be blocked on esp-println anyway. While a similar PR is already in esp-wifi, I'll need to hammer it some more still.